File: scoring-service/src/main/java/com/athena/creditscore/model/AdminUser.java
package com.athena.creditscore.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "admin_users")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminUser {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String username;

    @Column(name = "password_hash", nullable = false)
    private String passwordHash;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column
    private String email;

    /**
     * Role: ADMIN | ANALYST | VIEWER | CREDIT_RISK
     * Matches CHECK constraint in schema.sql
     */
    @Column(nullable = false)
    private String role;

    @Column(name = "totp_secret")
    private String totpSecret;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private boolean active = true;
}

File: user-service/internal/model/admin_user.go
package model

// AdminUser maps to the admin_users table.
// Uses password_hash column (bcrypt), separate from the users table.
type AdminUser struct {
	ID           int64  `gorm:"primaryKey;autoIncrement" json:"id"`
	Username     string `gorm:"uniqueIndex;not null" json:"username"`
	PasswordHash string `gorm:"column:password_hash;not null" json:"-"`
	FirstName    string `gorm:"column:first_name" json:"firstName"`
	LastName     string `gorm:"column:last_name" json:"lastName"`
	Email        string `json:"email"`
	Role         string `gorm:"not null" json:"role"`
	TotpSecret   string `gorm:"column:totp_secret" json:"-"`
	Active       bool   `gorm:"column:is_active;not null;default:true" json:"active"`
}

func (AdminUser) TableName() string { return "admin_users" }
